### PR TITLE
En_Light

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Light/z_en_light.c [hunk 1]
```diff
@@ -28,16 +30,247 @@ const ActorInit En_Light_InitVars = {
...
 };
 
-#endif
+typedef struct {
+    /* 0x00 */ Color_RGBA8 unk_00;
+    /* 0x04 */ Color_RGB8 unk_04;
+    /* 0x07 */ u8 unk_07;
+} EnLightStruct; // size = 0x8
+
+EnLightStruct D_808666D0[] = {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Light/z_en_light.c:39]: Remove appropriate trailing commas to make this format more sensibly.

File: src/overlays/actors/ovl_En_Light/z_en_light.c [hunk 2]
```diff
@@ -28,16 +30,247 @@ const ActorInit En_Light_InitVars = {
...
+    if (!ENLIGHT_GET_4000(&this->actor)) {
+        EnLightStruct* sp28 = &D_808666D0[ENLIGHT_GET_F(&this->actor)];
+        f32 temp_f2 = (Rand_ZeroOne() * 0.5f) + 0.5f;
+
+        Lights_PointSetColorAndRadius(&this->lightInfo, (u8)(sp28->unk_00.r * temp_f2), (u8)(sp28->unk_00.g * temp_f2),
+                                      (u8)(sp28->unk_00.b * temp_f2),
+                                      (this->actor.params < 0)         ? 100
+                                      : ENLIGHT_GET_2000(&this->actor) ? 730
+                                                                       : 300);
+    }
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Light/z_en_light.c:184]: I assume all this is required to match.

Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Light/z_en_light.c:184]: Yup unfortunately. Very messy.

Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Light/z_en_light.c:184]: I pulled the ternary out into a separate variable, but it seemingly has to be a ternary at least. Taking it out of the call hopefuly helps a bit.

Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Light/z_en_light.c]: Yeah, that is a significant improvement. You can call it `radius` to match the function arg, too.

File: src/overlays/actors/ovl_En_Light/z_en_light.c [hunk 3]
```diff
@@ -28,16 +30,173 @@ const ActorInit En_Light_InitVars = {
...
+}
+
+void func_80865F38(Actor* thisx, GlobalContext* globalCtx) {
+    EnLight* this = THIS;
+    EnLightStruct* sp38 = &D_808666D0[ENLIGHT_GET_F(&this->actor)];
+    f32 temp_f2;
+    f32 sp30 = this->actor.scale.x / (sp38->unk_07 * 0.0001f);
+    s32 sp2C = false;
+
+    if ENLIGHT_GET_1000 (&this->actor) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Light/z_en_light.c:126]:
```suggestion
    if (ENLIGHT_GET_1000 (&this->actor)) {
```

File: src/overlays/actors/ovl_En_Light/z_en_light.c [hunk 4]
```diff
@@ -28,16 +30,173 @@ const ActorInit En_Light_InitVars = {
...
+}
+
+void func_80865F38(Actor* thisx, GlobalContext* globalCtx) {
+    EnLight* this = THIS;
+    EnLightStruct* sp38 = &D_808666D0[ENLIGHT_GET_F(&this->actor)];
+    f32 temp_f2;
+    f32 sp30 = this->actor.scale.x / (sp38->unk_07 * 0.0001f);
+    s32 sp2C = false;
+
+    if (ENLIGHT_GET_1000 (&this->actor)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Light/z_en_light.c:126]:
```suggestion
    if (ENLIGHT_GET_1000(&this->actor)) {
```

**Review by hensldm — APPROVED** [review]
Looks like it needs a formatting.

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**